Source/EventParser.swift

```diff
@@ -64,15 +70,20 @@ class EventParser {
     }
 
     private func dispatchEvent() {
-        guard !data.isEmpty
-        else { return }
-        let messageEvent = MessageEvent(data: data, lastEventId: lastEventId)
         if let lastEventId = lastEventId {
             connectionHandler.setLastEventId(lastEventId)
         }
-        handler.onMessage(eventType: eventType, messageEvent: messageEvent)
+        guard !data.isEmpty
+        else {
+            eventType = ""
+            return
+        }
+        // remove the last LF
+        _ = data.popLast()
+        let messageEvent = MessageEvent(data: data, lastEventId: lastEventId)
+        handler.onMessage(eventType: eventType.isEmpty ? "message" : eventType, messageEvent: messageEvent)
```

Comment:
I believe this may also have been a change in the spec. Not allowing an empty event type field, so event a line like
```
event:
```
Does not change the event type from the default "message".

Reply:
I can't say whether this was "a change in the spec" because I don't have access to whatever spec you were originally working from, but it is true that "message" is always the default if `event:` was either omitted or empty. The relevant language in [the spec](https://html.spec.whatwg.org/multipage/server-sent-events.html#event-stream-interpretation) (btw, in the future it'd be good to link to the spec in the PR description so reviewers can check your interpretation) is:

> Initialize _event_'s `type` attribute to `message`, its `data` attribute to `data`, its _origin_ attribute to the serialization of the origin of the event stream's final URL (i.e., the URL after redirects), and its `lastEventId` attribute to the last event ID string of the event source.
> If the _event type_ buffer has a value other than the empty string, change the type of the newly created _event_ to equal the value of the _event type_ buffer.

What you're calling "the event type" in this comment is, I believe, the same thing that the spec calls "the event type buffer".